backend/app/api/routers/threads.py

```diff
@@ -0,0 +1,68 @@
+from datetime import datetime
+from http import HTTPStatus
+from typing import Optional
+
+from fastapi import APIRouter, WebSocket, WebSocketDisconnect
+
+from app.api.exceptions import EntityNotFound
+from app.api.managers import WsConnectionManager
+from app.api.schemas import Filter, MessageCreate, Sender, SupportCreate, SupportThread
+from app.api.services import SupportThreadService
+
+router = APIRouter()
+ws_manager = WsConnectionManager()
+
+
+@router.post('/threads', response_model=SupportThread, status_code=HTTPStatus.CREATED)
+async def create_thread(thread: SupportCreate) -> SupportThread:
+    thread_new = await SupportThreadService.create(thread)
+
+    return thread_new
+
+
+@router.get('/threads', response_model=list[SupportThread])
+async def find_all_threads(flt: Optional[Filter] = None) -> list[SupportThread]:
+    threads = await SupportThreadService.find_all(flt)
+
+    return threads
+
+
+@router.get('/threads/{thread_id}', response_model=SupportThread)
+async def find_thread(thread_id: int) -> SupportThread:
+    thread = await SupportThreadService.find_by_id(thread_id)
+
+    if not thread:
+        raise EntityNotFound('thread')
+
+    return thread
+
+
+@router.put('/threads/{thread_id}', response_model=SupportThread)
+async def update_thread(thread_id: int, thread: SupportCreate) -> SupportThread:
+    thread_upd = await SupportThreadService.update(thread, thread_id)
+
+    return thread_upd
+
+
+@router.delete('/threads/{thread_id}')
+async def delete_thread(thread_id: int) -> None:
+    await SupportThreadService.delete(thread_id)
+
+
+@router.websocket('/ws/{thread_id}')
+async def websocket_endpoint(websocket: WebSocket, thread_id: int) -> None:
+    await ws_manager.connect(websocket, room_id=thread_id)
+    try:
+        while True:
+            data = await websocket.receive_text()
+            message = MessageCreate(
+                created_at=datetime.utcnow(), content=data, sender=Sender.client
+            )  # TODO: make sender recognizer
```

Comment:
So as I remember we decided to add opportunity to recover history, so every message has to contain sender to be correctly proceed on the frontend part. Since we do not have identification service yet, there were 2 options, first is to set sender field in pydantic schema as optional, second is to put some value of Sender(Enum). I chose the second approach, and now I understand it's confusing, and I set the sender field as optinal